### ISSUE TITLE
Embargo date validation against data publication

### ISSUE THREAD

**Issue by kitchenprinzessin3880** [AuScope/ckan-docker#66]
Implementation changes required

1. For a given dataset, if there is an embargo date, when an editor or admin changes the data from private to public, this should not be allowed.
2. Editor/Admin cannot change data from public to private to comply with the DOI practice.

**Comment by stuartwoodman** [comment]
This should be fairly straight forward. The class validation.py has a visibility_validation method that prevents a member from changing the visibility field to public (only admins or editors), it will just need an additional check to confirm that the current date is greater than or equal to the embargo date if this has been set.

**Comment by kitchenprinzessin3880** [comment]
@stuartwoodman is there a way we can set max embargo period? 1 year from the date of deposit..or is this related to UI work? @NTaherifar

**Comment by stuartwoodman** [comment]
> @stuartwoodman is there a way we can set max embargo period? 1 year from the date of deposit..or is this related to UI work? @NTaherifar

We could achieve this by validating the input when the user submits, but it would still allow them to select a date from the widget > 1 year.

I have found an issue with the existing embargo validation, I've got code now that won't let you publish a record if the embargo date is in the future, but when testing this today I clicked Publish and it told me that the embargo date must be in the future, forcing me to manually delete the embargo date and then publish. Which may end up being the easiest way to go, I'm not sure how we deal with it otherwise, except for some convoluted method of checking for an expired embargo date when the record is first edited and if found, then delete it. That might work.
